server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -984,7 +994,8 @@ public Object[] addGeneratedValues(IndexItem item) {
                     root = new HashMap<>();
                     extendedValues.add(root);
                 } else {
-                    root = (Map<String, Object>) insertValues[valueIdx];
+                    assert valueIdx < extendedValues.size() : "Number of values being sent to replica must match number of targets in replica request";
+                    root = (Map<String, Object>) extendedValues.get(valueIdx);
```

Comment:
for sub-columns root can be null. 
If it's okay I want to follow up on this in another PR

Reply:
so, with this PR it's NPE, on master it's `IndexOutOfBounds` and on state before https://github.com/crate/crate/pull/15272 we couldn't store object with non-determisnitic sub-column it root is not in targets (also out of bounds but with -1, on master it's some positive index)